### PR TITLE
Generate self-contained musl binary

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

Compile x86_64-musl binary as static position independent executable. A static position independent executable is similar to static executable, but can be loaded at any address without a dynamic linker.

As a test, I downloaded the generated `pihole-FTL-musl-linux-x86_64` binary into a Debian-based `x86_64` container and made it executable. Thereafter, ran our test suite* using this binary and all tests succeeded.

---
*) For running the tests, 
- cloned the `FTL` repository, 
- checked out the current `development` branch,
- copied the previously downloaded binary into the cloned directory, and
- ran `test/run.sh`.

I did *not* build the binary in the cloned repository.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -98,11 +97,19 @@ CCFLAGS=-std=gnu11 -I$(IDIR) $(WARNFLAGS) -D_FILE_OFFSET_BITS=64 $(HARDENING_FLA
...
 # for dnsmasq we need the nettle crypto library and the gmp maths library
 # We link the two libraries statically. Although this increases the binary file size by about 1 MB, it saves about 5 MB of shared libraries and makes deployment easier
-#LIBS=-pthread -lnettle -lgmp -lhogweed
-LIBS=-pthread -lrt -Wl,-Bstatic -L/usr/local/lib -lhogweed -lgmp -lnettle -Wl,-Bdynamic
+LIBS=-pthread -lrt -Wl,-Bstatic -L/usr/local/lib -lhogweed -lgmp -lnettle
 # Flags for compiling with libidn : -lidn
 # Flags for compiling with libidn2: -lidn2
 
+# Do we want to compile a statically linked musl executable?
+ifeq "$(CIRCLE_JOB)" "x86_64-musl"
```
Comment (Mcat12) [inline Makefile]: Can this be a more general check (not tied to the CI)? Something like checking if `MUSL` equals `true`?

File: log.c [hunk 2]
```diff
@@ -140,7 +140,7 @@ void log_counter_info(void)
 void log_FTL_version(const bool crashreport)
 {
 	logg("FTL branch: %s", GIT_BRANCH);
-	logg("FTL version: %s", GIT_TAG);
+	logg("FTL version: %s", GIT_VERSION);
```
Comment (Mcat12) [inline log.c]: Is this meant for #595?

File: Makefile (hunk 1 above)
Comment (Mcat12) [reply Makefile]: Actually, doesn't need to be tied to musl, the variable could be named `STATIC`.

Comment (DL6ER) [reply Makefile]: I had the same idea initially. For the same of simplicity (use the same build job for all architectures), we'd realize this via an environment variable set in the musl docker container. What do you think about this?

File: log.c (hunk 2 above)
Comment (DL6ER) [reply log.c]: Yes, this got committed by accident, I'll remove it from this PR.

File: .circleci/config.yml [hunk 3]
```diff
@@ -13,8 +13,7 @@ version: 2
         name: "Build"
         command: |
           BRANCH=$([ -z "$CIRCLE_TAG" ] && echo "$CIRCLE_BRANCH" || echo "master")
-
-          make GIT_BRANCH="${BRANCH}" GIT_TAG="${CIRCLE_TAG}"
+          make GIT_BRANCH="${BRANCH}" GIT_TAG="${CIRCLE_TAG}" STATIC="${STATIC}"
```
Comment (DL6ER) [inline .circleci/config.yml]: I added it here to be explicit. As `STATIC` is an environment variable, we can also leave it off as `make` will silently pick it up.

File: Makefile (hunk 1 above)
Comment (Mcat12) [reply Makefile]: To make it easier to change the value, we should instead set the variable in the CircleCI environment variables for that job.

Comment (DL6ER) [reply Makefile]: I intentionally didn't choose this path as it again ties us more to the CI. The sole purpose of the `musl` container is to provide a self-contained binary. The container should behave similarly on the CI and when compiling local. This wouldn't be the case if we add in a (hidden) extra environment variable in the CI.

**Pull request merged**